src/components/tabs/tab-title/tab-title.spec.js

```diff
@@ -0,0 +1,288 @@
+import React from 'react';
+import TestRenderer from 'react-test-renderer';
+import 'jest-styled-components';
+import { shallow } from 'enzyme';
+import TabTitle from './tab-title.component';
+import StyledTabTitle from './tab-title.style';
+import classicTheme from '../../../style/themes/classic';
+import { assertStyleMatch } from '../../../__spec_helper__/test-utils';
+import baseTheme from '../../../style/themes/base';
+
+function render(props) {
+  return shallow(
+    <TabTitle
+      title='Tab Title 1'
+      dataTabId='uniqueid1'
+      { ...props }
+    />
+  );
+}
+
+function renderStyles(props) {
+  return TestRenderer.create(
+    <StyledTabTitle
+      title='Tab Title 1'
+      dataTabId='uniqueid1'
+      { ...props }
+    />
+  );
+}
+
+describe('TabTitle', () => {
+  let wrapper;
+  it('renders as expected', () => {
+    wrapper = renderStyles();
+    expect(wrapper.toJSON()).toMatchSnapshot();
+  });
+
+  it('renders a title as its child with a text passed as a prop', () => {
+    wrapper = render();
+    expect(wrapper.children()).toHaveLength(1);
+    expect(wrapper.children().text()).toEqual('Tab Title 1');
+  });
+
+  it('contains custom className if passed as a prop', () => {
+    wrapper = render({ className: 'class' });
+    expect(wrapper.find('.class').exists()).toEqual(true);
+  });
+
+  describe('when clicked', () => {
+    it('triggers onClick function', () => {
+      const onClickFunction = jest.fn();
+      wrapper = render({ onClick: onClickFunction });
+      wrapper.simulate('click');
+      expect(onClickFunction).toHaveBeenCalled();
+    });
+  });
+
+  describe('attributes', () => {
+    wrapper = render();
+    it('role equals "tab"', () => {
+      expect(wrapper.find("[role='tab']").exists()).toEqual(true);
+    });
+    it('data-element equals "select-tab"', () => {
+      expect(wrapper.find("[data-element='select-tab']").exists()).toEqual(true);
+    });
+    it('data-tabid equals tabId', () => {
+      expect(wrapper.find("[data-tabid='uniqueid1']").exists()).toEqual(true);
+    });
+  });
+
+  describe('when position prop is set to left', () => {
+    it('applies proper styles', () => {
+      wrapper = renderStyles({ position: 'left' });
+      assertStyleMatch(
+        {
+          backgroundColor: 'transparent',
+          borderRight: `2px solid ${baseTheme.disabled.background}`,
+          display: 'block',
+          height: 'auto',
+          marginLeft: '0px',
+          marginTop: '2px'
+        },
+        wrapper.toJSON()
```

Comment:
Do we  need a `toJSON` function?

Reply:
yes, otherwise I got an error: ' No style rules found on passed Component'